src/backend/vulkan/src/lib.rs

```diff
@@ -128,19 +175,39 @@ impl Instance {
                 p_next: ptr::null(),
                 flags: vk::InstanceCreateFlags::empty(),
                 p_application_info: &app_info,
-                enabled_layer_count: 0,
-                pp_enabled_layer_names: ptr::null(),
-                enabled_extension_count: str_pointers.len() as u32,
-                pp_enabled_extension_names: str_pointers.as_ptr(),
+                enabled_layer_count: layers.len() as _,
+                pp_enabled_layer_names: str_pointers.as_ptr(),
```

Comment:
What will happen here if the layers are not present? Can we add checks here to filter out some layers?